src/radio.cpp

```diff
@@ -100,18 +100,18 @@ void zmq::radio_t::xwrite_activated (pipe_t *pipe_)
 
 void zmq::radio_t::xpipe_terminated (pipe_t *pipe_)
 {
-    for (subscriptions_t::iterator it = subscriptions.begin (); it != subscriptions.end (); ) {
```

Comment:
So... Standard lies? ;) 

Erase returns iterator following the last removed element. If the iterator pos refers to the last element, the end() iterator is returned.